src/Graviton/CoreBundle/Composer/ScriptHandlerBase.php

```
@@ -0,0 +1,98 @@
+     * to forthcoming listeners.
+     */
+    protected static $options = array(
+        'symfony-app-dir' => 'app',
+    );
+
+    /**
+     * Returns some options - adapted from Sensio DistributionBundle Command
+     *
+     * @param CommandEvent $event Event
+     *
+     * @return array Options
+     */
+    protected static function getOptions(CommandEvent $event)
+    {
+        $options = array_merge(self::$options, $event->getComposer()->getPackage()->getExtra());
+        $options['process-timeout'] = $event->getComposer()->getConfig()->get('process-timeout');
+        return $options;
+    }
+
+    /**
+     * Executes a command
+     *
+     * @param CommandEvent $event      Event
+     * @param string       $consoleDir Console dir
+     * @param string       $cmd        Command
+     * @param int          $timeout    Timeout
+     *
+     * @return void
+     */
+    protected static function executeCommand(CommandEvent $event, $consoleDir, $cmd, $timeout = 300)
+    {
+        $php = escapeshellarg(self::getPhp(false));
+        $console = escapeshellarg($consoleDir.'/console');
+        if ($event->getIO()->isDecorated()) {
+            $console .= ' --ansi';
+        }
+
+        $process = new Process($php.' '.$console.' '.$cmd, null, null, null, $timeout);
+        $process->run(
+            function ($type, $buffer) use ($event) {
+                $event->getIO()->write($buffer, false);
+            }
+        );
+
+        if (!$process->isSuccessful()) {
+            throw new \RuntimeException(
+                sprintf(
+                    'An error occurred when executing the "%s" command.',
+                    escapeshellarg($cmd)
+                )
+            );
+        }
+    }
+
+
+    /**
+     * Finds the path to the php executable
+     *
+     * @param bool $includeArgs include args
+     *
+     * @return bool|false|null|string path
+     */
+    protected static function getPhp($includeArgs = true)
```

I was also mentioning this since this is the first time that these methods are being used by a subclass and therefore expose an internal API that, in theory, needs testing.

I'm letting it go due to the fact that these get executed on each travis build anyway and will fail early and hard if they really have issues.